shell/platform/darwin/ios/ios_surface_gl.mm

```diff
@@ -38,8 +38,16 @@
 std::unique_ptr<Surface> IOSSurfaceGL::CreateGPUSurface(GrDirectContext* gr_context) {
   if (gr_context) {
     return std::make_unique<GPUSurfaceGL>(sk_ref_sp(gr_context), this, true);
+  } else {
+    IOSContextGL* gl_context = CastToGLContext(GetContext());
+    sk_sp<GrDirectContext> context = gl_context->GetMainContext();
+    if (!context) {
+      context = GPUSurfaceGL::MakeGLContext(this);
+      gl_context->SetMainContext(context);
```

Comment:
Ditto here. Seems like we have 2 timing characteristics. The metal case just makes this GrDirectContext way earlier when creating the shell's on_create_platform_view, whereas this makes it way later in FlutterViewController's viewWillAppear. There could be minutes in between if the engine was prewarmed. 

I'm a bit worried that there might be side-effects involving platform views or memory etc that are hard to reproduce due to the rendering API differences. 

It doesn't have to be in this PR, but I suppose this way works better memory wise. Can we make metal work this way too?